go/private/context.bzl

```diff
@@ -280,6 +283,19 @@ def _collect_runfiles(go, data, deps):
         runfiles = runfiles.merge(get_source(t).runfiles)
     return runfiles
 
+def _collect_cc_infos(deps, cdeps):
+    cc_infos = []
+    for dep in cdeps:
+        if CcInfo in dep:
+            cc_infos.append(dep[CcInfo])
+    for dep in deps:
+        # dep may be a struct, which doesn't support indexing by providers.
+        if is_struct(dep):
```

Comment:
Just so I understand, can you give an example of this?

Reply:
The deps may be providers rather than targets, see the note on `attr` at https://github.com/bazelbuild/rules_go/blob/master/go/toolchains.rst#library_to_source.